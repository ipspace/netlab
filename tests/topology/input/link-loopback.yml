# Tests the 'loopback' link type (#413 / #708)
#
defaults:
  links.stub_loopback: True
  devices.iosv.features.stub_loopback: False

nodes:
  r1:
    device: eos
  r2:
    device: iosv
  h1:
    device: linux

links:
- h1:                       # Linux does not support loopback interfaces, should remain a stub network
- r1:                       # Arista EOS supports loopback interfaces, should be converted into a loopback
- r2:                       # Stub loopbacks were disabled on IOSv, this should remain a regular stub network
- r2:
    ipv4: 10.1.0.1/32       # Force a /32 as the node IPv4 address
  type: loopback            # While this one is an explicit loopback
- r1:
  prefix: 10.1.0.2/32       # Use a /32 prefix on an implicit loopback interface
- r1:
  pool: vrf_loopback        # Take a prefix from the loopback pool
