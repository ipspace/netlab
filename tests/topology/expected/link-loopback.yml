input:
- topology/input/link-loopback.yml
- package:topology-defaults.yml
links:
- bridge: input_1
  interfaces:
  - ifindex: 1
    ifname: eth1
    ipv4: 172.16.0.3/24
    node: h1
  linkindex: 1
  node_count: 1
  prefix:
    ipv4: 172.16.0.0/24
  type: lan
- interfaces:
  - ifindex: 10001
    ifname: Loopback1
    ipv4: 172.16.1.1/24
    node: r1
  linkindex: 2
  node_count: 1
  prefix:
    ipv4: 172.16.1.0/24
  type: loopback
- bridge: input_3
  interfaces:
  - ifindex: 1
    ifname: GigabitEthernet0/1
    ipv4: 172.16.2.2/24
    node: r2
  linkindex: 3
  node_count: 1
  prefix:
    ipv4: 172.16.2.0/24
  type: stub
- interfaces:
  - ifindex: 10001
    ifname: Loopback1
    ipv4: 10.1.0.1/32
    node: r2
  linkindex: 4
  node_count: 1
  prefix:
    ipv4: 172.16.3.0/24
  type: loopback
- interfaces:
  - ifindex: 10002
    ifname: Loopback2
    ipv4: 10.1.0.2/32
    node: r1
  linkindex: 5
  node_count: 1
  prefix:
    ipv4: 10.1.0.2/32
  type: loopback
- interfaces:
  - ifindex: 10003
    ifname: Loopback3
    ipv4: 10.2.0.1/32
    node: r1
  linkindex: 6
  node_count: 1
  pool: vrf_loopback
  prefix:
    ipv4: 10.2.0.1/32
  type: loopback
name: input
nodes:
  h1:
    af:
      ipv4: true
    box: generic/ubuntu2004
    device: linux
    id: 3
    interfaces:
    - bridge: input_1
      ifindex: 1
      ifname: eth1
      ipv4: 172.16.0.3/24
      linkindex: 1
      name: h1 -> stub
      neighbors: []
      type: lan
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.103
      mac: 08:4f:a9:00:00:03
    name: h1
    role: host
  r1:
    af:
      ipv4: true
      ipv6: true
    box: arista/veos
    device: eos
    id: 1
    interfaces:
    - ifindex: 10001
      ifname: Loopback1
      ipv4: 172.16.1.1/24
      linkindex: 2
      name: r1 -> stub
      neighbors: []
      type: loopback
      virtual_interface: true
    - ifindex: 10002
      ifname: Loopback2
      ipv4: 10.1.0.2/32
      linkindex: 5
      name: r1 -> stub
      neighbors: []
      type: loopback
      virtual_interface: true
    - ifindex: 10003
      ifname: Loopback3
      ipv4: 10.2.0.1/32
      linkindex: 6
      name: r1 -> stub
      neighbors: []
      pool: vrf_loopback
      type: loopback
      virtual_interface: true
    loopback:
      ipv4: 10.0.0.1/32
      ipv6: 2001:cafe::1/64
    mgmt:
      ifname: Management1
      ipv4: 192.168.121.101
      mac: 08:4f:a9:00:00:01
    name: r1
  r2:
    af:
      ipv4: true
    box: cisco/iosv
    device: iosv
    id: 2
    interfaces:
    - bridge: input_3
      ifindex: 1
      ifname: GigabitEthernet0/1
      ipv4: 172.16.2.2/24
      linkindex: 3
      name: r2 -> stub
      neighbors: []
      type: stub
    - ifindex: 10001
      ifname: Loopback1
      ipv4: 10.1.0.1/32
      linkindex: 4
      name: r2 -> stub
      neighbors: []
      type: loopback
      virtual_interface: true
    loopback:
      ipv4: 10.42.42.1/32
    mgmt:
      ifname: GigabitEthernet0/0
      ipv4: 192.168.121.102
      mac: 08:4f:a9:00:00:02
    name: r2
provider: libvirt
